### PR TITLE
Change default labels layer to uint8.

### PR DESCRIPTION
**Pull request by Czaki**
This PR changes the default type for labels created with the new labels button to `uint8`.

Currently, it is `int` that defaults to `int64` on lLnux and MacOs and `int32` on Windows.

During development of #6467 I have found that defaulting to `int` may create some performance problems. Because of this, I suggest this change.

This change reduces memory usage four times on Windows and eight times on Linux and MacOS. 

The user still could change the type of labels layer by right click menu.

### CODE REVIEW

**Comment by Czaki** [comment]
Fast check do not show part of docs that needs to be updated, but maybe I miss something.

**Comment by jni** [comment]
@Czaki while thinking about this I had the following idea: when people make a new layers label in this way, it is typically to paint labels, and usually that means very small values. They could almost always fit in uint8. So my suggestion is:

- we change to uint8 by default
- we only update the dtype when the user selects a negative label or a larger-than-dtype label in the selection box. (Potentially, we can display a warning when that happens.)

Ultimately, I am very supportive of this change. We've seen some very significant performance issues with higher dtypes, and the choice to use int dates back to a time when users didn't have a quick way to change the data type of a layer. Now, it is just a right-click away, so I think we should use a small dtype to begin with and expand just-in-time.

**Comment by psobolewskiPhD** [comment]
> But we could just update the documentation to highlight right click menu?

You mean this? 
https://napari.org/dev/tutorials/fundamentals/viewer.html#right-click-menu
(note I just noticed the fix to the Change dtype got over-written by my other merge. I've made a PR)

Maybe worth mentioning dtype here too https://napari.org/dev/howtos/layers/labels.html

**Comment by Czaki** [comment]
> You mean this?

yes. To increase visibility

**Comment by Czaki** [comment]
> The button uses this method afaik:

I even modify it here. 
The question is how to distinguish such created label and suggest change type, and labels added in different ways to not suggest then.

**Comment by brisvag** [comment]
Sorry, answered quickly without reading further. You could use the layer source? I don't know if we have a "button" source specifically, but any plugin-created layer should have a different source.

**Comment by jni** [comment]
Yeah, source is a good idea! (I was going to suggest unstructured metadata but source is so much better! 😂)

**Comment by Czaki** [comment]
After rethink and check the code. Did we should do both changes in one PR? 

We currently set limits of label selectors based on data type. The PR for proper handling conversion will not be small. 

Maybe we could just make a small PR to reduce the default type, to be cherry-picked to 0.4.19, and make a separate PR with a convert suggestion somehow later?

**Comment by Czaki** [comment]
THe Source is defined here:

https://github.com/napari/napari/blob/dbafdd08e0202bf78fcec9dcc73ead6742a2cf39/napari/layers/_source.py#L14-L36

I do not see a proper field to use. And do not know how to name new generic field.

**Comment by jni** [comment]
Yeah I'm happy to do two (or more) different PRs. But then I would do the following PRs:

- change default to uint8 (this one)
- warn when trying to change selected label to out-of-range value, and warning points out how to convert layer dtype using right-click
- run the dtype change automatically. (another idea: this could happen on the second click, after the warning)
- right click on the new labels button has menu to select new labels dtype

**Comment by Czaki** [comment]
Ok. I have updated dtype to `uint8`. I have also fixed bug in Qt layer controls to update the range of label selectors with and add a test for such a scenario.

**Review by Carreau — APPROVED** [review]

File: napari/_qt/layer_controls/_tests/test_qt_labels_layer.py [hunk 1]
```diff
@@ -115,3 +115,16 @@ def test_preserve_labels_checkbox(make_labels_controls):
     assert not layer.preserve_labels
     qtctrl.preserveLabelsCheckBox.setChecked(True)
     assert layer.preserve_labels
+
+
+def test_change_label_selector_range(make_labels_controls):
+    """Tests that changing the label selector range updates the layer."""
```
Comment (jni) [inline napari/_qt/layer_controls/_tests/test_qt_labels_layer.py]:
```suggestion
    """Changing the label layer dtype should update label selector range."""
```

If I'm reading the test correctly, this doesn't check that updating the selector range updates the layer, but rather the opposite, right?

**Review by jni — APPROVED** [review]
Other than a minor docstring update suggestion, I think this is ready to go!

However, I don't think this is an urgent fix @Czaki. And it would be good to release it together with some of the other fixes we have discussed (e.g. warnings when trying out-of-range values), so my suggestion is to put the 0.5.0 milestone on it.

**Comment by Czaki** [comment]
I have created #6479 which should be merged before this.

**Comment by psobolewskiPhD** [comment]
> I have created #6479 which should be merged before this.

I think you can do `Depends on #6479` in the OP? Or does that only work in the docs repo

**Comment by Czaki** [comment]
> I think you can do `Depends on #6479` in the OP? Or does that only work in the docs repo

It works only in docs repo, we may think to add it the main repo also

**Comment by psobolewskiPhD** [comment]
I think it makes sense if we want big PRs broken down into smaller chunks. Makes the dependency of them easier.

**Comment by jni** [comment]
Ok, #6479 has now been merged so I've merged in main and added the ready-to-merge label here. Since this has already been sitting for a while I suggest that it can be merged as soon as all the checks are green. Thanks @Czaki!

**Review by Carreau — APPROVED** [review]

**Review by brisvag — APPROVED** [review]

**Pull request merged**